### PR TITLE
Use Maven Spotless plugin in postprocessor to handle all code formatting

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR removes usage of Google format in favor of having the postprocessor using Maven Spotless plugin with an Eclipse format configuration perform formatting. This removes our managed dependency and uses a format configuration that could be shared so that all formatting is the same.

### CODE REVIEW

File: fluentgen/pom.xml [hunk 1]
```diff
@@ -40,12 +40,6 @@
       <version>1.0.0-beta.1</version>
     </dependency>
 
-    <dependency>
-      <groupId>com.google.googlejavaformat</groupId>
-      <artifactId>google-java-format</artifactId>
-      <version>1.17.0-beta.1</version>
-    </dependency>
```
Comment (weidongxu-microsoft) [inline fluentgen/pom.xml:47]: fluent does not use postprocess nor Eclipse language server.

Maybe we first keep it here, and let @XiaofeiCao do the migration later.

I am a bit worried about the extra time when invoking Eclipse language server (it's pretty slow).

File: postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java [hunk 2]
```diff
@@ -118,7 +99,7 @@ public boolean processInternal() {
             } else if (className.startsWith("src") && className.endsWith(".java")) {
                 customizationClass = loadCustomizationClassFromJavaCode(className, getBaseDirectory(), logger);
             } else {
-                customizationClass = loadCustomizationClassFromReadme(className, readme);
+                throw new RuntimeException("Unable to load customization class.");
```
Comment (weidongxu-microsoft) [inline postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java:103]: why this change about readme?

File: postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java [hunk 3]
```diff
@@ -321,54 +286,28 @@ private static void attemptMavenInstall(Path pomPath, Logger logger) {
...
-            logger.info("Unable to determine Java version to verify if Java 11+ is being used, which is the "
-                + "requirement to run Autorest code customizations.");
-            return -1;
+        try {
+            Process process = new ProcessBuilder(command)
+                .redirectOutput(ProcessBuilder.Redirect.DISCARD)
+                .redirectError(ProcessBuilder.Redirect.DISCARD)
+                .start();
+            process.waitFor(30, TimeUnit.SECONDS);
+            process.exitValue();
```
Comment (weidongxu-microsoft) [inline postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java:306]: why return value ignored?

File: typespec-extension/src/main/java/com/azure/typespec/Main.java
Comment (weidongxu-microsoft) [inline typespec-extension/src/main/java/com/azure/typespec/Main.java:1]: typespec-java does not pass postprocess like autorest does.

Currently it would call here https://github.com/Azure/autorest.java/blob/main/typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java#L144-L149, which seems won't call the spotless code.

**Comment by weidongxu-microsoft** [comment]
For fluent CI, I think it fails due to https://github.com/Azure/autorest.java/pull/2364#discussion_r1371032869

For typespec CI, you first need to move them to Java 11 as well https://github.com/alzimmermsft/autorest.java/blob/UseSpotlessInPostProcessor/eng/pipelines/ci-typespec-java.yaml#L30
(nightly-build CI as well)
Then this issue https://github.com/Azure/autorest.java/pull/2364#discussion_r1371063630

**Comment by haolingdong-msft** [comment]
@alzimmermsft  May I know the benefit of this change, is it due to https://github.com/Azure/autorest.java/issues/2321#issuecomment-1743477724 the javadoc issue? Is there any other benefits?

File: fluentgen/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply fluentgen/pom.xml:47]: Fortunately, Spotless doesn't require the Eclipse language server to perform Eclipse code formatting. Don't know the exact design but I know from investigating the removal of Google Format that Eclipse JDT Core has the ability to run code formatting without use of the language server and is just as fast as Google Format. Though, using Spotless requires code to be on disk rather than in memory, which will be slower but shouldn't be by much.

Based on that, thoughts on copying the code I added into postprocessor to fluent, and leaving the Google Format dependency out.

File: postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java (hunk 2 above)
Comment (alzimmermsft) [reply postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java:103]: This is removing the ability to configure code customizations using a Java code block in the README. This is pushing the requirement to use a proper Java source code file or precompiled JAR for code customizations.

File: fluentgen/pom.xml (hunk 1 above)
Comment (weidongxu-microsoft) [reply fluentgen/pom.xml:47]: Yeah, we can just copy the code to fluentgen.

Maybe another solution is to put the spotless related code to a module that can be used independently from postprocess (like current "partial-update" module). Anyway if we really want it we can always refactor later.

File: postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply postprocessor/src/main/java/com/azure/autorest/postprocessor/Postprocessor.java:103]: Got it. This feature is likely rarely used.

**Comment by weidongxu-microsoft** [comment]
> @alzimmermsft May I know the benefit of this change, is it due to [#2321 (comment)](https://github.com/Azure/autorest.java/issues/2321#issuecomment-1743477724) the javadoc issue? Is there any other benefits?

One thing is about the google java format. It is built from our own fork (and we actually built 2 of them :-)) and never released to maven. Using package outside of maven could be a security risk.

File: fluentgen/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply fluentgen/pom.xml:47]: I would save the larger refactor for later as there is other functionality copied across the plugins which would be useful to centralize.

**Comment by alzimmermsft** [comment]
Closing in favor of #2369 as it shows the code changes plus on regeneration change to make the review easier.
